test/gmp.jl

```diff
@@ -441,8 +441,25 @@ end
 @test isqrt(big(4)) == 2
 @test isqrt(big(5)) == 2
 
-@test big(5)^true == big(5)
-@test big(5)^false == one(BigInt)
+
+@testset "Exponentiation operator" begin
+    @test big(5)^true == big(5)
+    @test big(5)^false == one(BigInt)
+    testvals = Int8[-128:-126; -3:3; 125:127]
+    @testset "BigInt and Int8 are consistent: $i^$j" for i in testvals, j in testvals
+        int8_res = try
+            i^j
+        catch e
+            e
+        end
+        if int8_res isa Int8
+            @test (big(i)^big(j)) % Int8 === int8_res
+        else
+            # Test both have exception of the same type
+            @test_throws typeof(int8_res) big(i)^big(j)
```

Comment:
According to my local tests, it is actually possible to check for `int8_res` directly, which would ensure the error message itself is also the same, helping ensure the error originated from the same location. Is there a reason you prefer only testing the type?

Reply:
The error messages for BigInt and Int8 are currently different. This is because Int8 uses `throw_domerr_powbysq(x, p)` while BigInt has a custom message.